### ISSUE TITLE
Unsupported client version: make user feedback more actionable

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/client#6262]
oC desktop sync client 2.4 has great new features and security with OAuth 2.0 support etc. oC admins might want to block older clients than 2.4. This is easy doable in the `config.php`:

```
  'minimum.supported.desktop.version' => '2.4.0',
```

But feedback for users is not very actionable:

![ie11_-_win8_1](https://user-images.githubusercontent.com/214010/34258951-95554c34-e660-11e7-9f06-209a9fcb98b8.png)

![pasted_image_21_12_17__15_07](https://user-images.githubusercontent.com/214010/34258993-be40865e-e660-11e7-850a-14fa030b61b1.png)

![ie11_-_win8_1_and_os_x_10_11](https://user-images.githubusercontent.com/214010/34259029-e943ffca-e660-11e7-9847-b5cb45253151.png)

![cursor_and_ie11_-_win8_1](https://user-images.githubusercontent.com/214010/34259019-de3bf3e4-e660-11e7-8e9a-35cbbd07ccca.png)

- Users don't know what is the required information
-  Users don't know where to get the required information

Any idea how to improve here?

(tested with oC 10.0.4 and client 2.3.4 and client 2.4.0 on 32-bit Win 8.1.)

**Comment by foobarable** [comment]
The user should at least see the followoing information:

- Version required 
- A link to the download page of the required version

**Comment by SamuAlfageme** [comment]
@michaelstingl the thing is, in order to do this properly - server should send an error code on the body of the 403 to tell apart different errors on authentication (e.g. account disabled, credentials invalid...). 

Raised https://github.com/owncloud/core/issues/29948

**Comment by SamuAlfageme** [comment]
@michaelstingl - In this particular case, we could hide the `to PROPFIND https://<server>/remote.php/webdav/` part; which might be the less clear from a user standpoint. What do you think?

**Comment by foobarable** [comment]
I agree, the remote.php/webdav part is not relevant for the user. Nonetheless, the server should give  more information on why an error occured so the client can display proper error messages

**Comment by ckamm** [comment]
@michaelstingl @SamuAlfageme For some errors like "client cert is needed" or "ip needs to be added as trusted domain" the server sends a html body in addition to the error code and the wizard will display that. I suggest that the server could do the same for the "unsupported client version" error - that message can then contain the minimum version and download links.

I don't think it's reliable to put download link information into the client because when this error occurs you will always have an older client interacting with a newer server - the information stored in the older client might very well be outdated.

**Comment by foobarable** [comment]
Actually, the client should fetch it's current configuration from the server so admins could push new configuration or defaults to all clients. This would ofc include the download links.
